pkg/auth/tokensource/tokensource.go

```diff
@@ -33,6 +36,7 @@ func (t *ReuseTokenSourceWithInvalidate) Token() (*oauth2.Token, error) {
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to get token")
 	}
+	token.Expiry = token.Expiry.Add(-t.expiryDelta)
```

Comment:
Hmm, what you should be doing is as base do [ReuseTokenSourceWithExpiry](https://cs.opensource.google/go/x/oauth2/+/v0.17.0:oauth2.go;l=400) in l.25 and then we can skip the line here and have it handled by the lib code.

Reply:
I don't think that would work. Reason being that then you cannot `Invalidate` the token anymore. Even if you request a fresh token from the base, `ReuseTokenSourceWithExpiry` will just give you the cached token. But we want to be able to generate a fresh token from the base when the underlying STS config has changed (we do this currently by calling `Invalidate`).